beacon-chain/blockchain/forkchoice/service.go

```diff
@@ -98,6 +103,34 @@ func (s *Store) GenesisStore(
 		return errors.Wrap(err, "could not save genesis state in check point cache")
 	}
 
+	if err := s.cacheGenesisState(ctx); err != nil {
+		return errors.Wrap(err, "could not cache initial sync state")
+	}
+
+	return nil
+}
+
+// This sets up gensis for initial sync state cache
+func (s *Store) cacheGenesisState(ctx context.Context) error {
+	if featureconfig.Get().InitSyncCacheState {
+		return nil
+	}
+
+	genesisState, err := s.db.GenesisState(ctx)
+	if err != nil {
+		return err
+	}
+	stateRoot, err := ssz.HashTreeRoot(genesisState)
```

Comment:
Stateutils.HashTreeRootState here

Reply:
Need to merge #4197 first

Reply:
Nvm

Reply:
Taked with @rauljordan , will make a subsequent PR to refactor block chain service and fork choice to adopt `HashTreeRootState`